src/main/kotlin/helpers.kt

```diff
@@ -118,9 +116,9 @@ fun Project.standardPublishing(pom: MavenPom.() -> Unit) {
     val deploymentPassword = (System.getenv("OSSRH_PASSWORD")?.takeUnless { it.isEmpty() }
         ?: props?.getProperty("ossrhPassword")?.toString())
         ?.trim()
-    val useDeployment = deploymentUser != null || deploymentPassword != null
+    val useDeployment = deploymentUser != null && deploymentPassword != null
 
-    sourceAndJavadoc()
+    sources(generateJavadoc = props?.getProperty("generateJavadoc")?.toBoolean() ?: true)
```

Comment:
generateJavadoc -> publishJavadoc in accordance with what the function actually does